lib/modules/manager/npm/post-update/pnpm.ts

```diff
@@ -79,8 +81,17 @@ export async function generateLockFile(
 
     // postUpdateOptions
     if (config.postUpdateOptions?.includes('pnpmDedupe')) {
-      logger.debug('Performing pnpm dedupe');
-      commands.push('pnpm dedupe');
+      const pnpmVersionFromPackageJson = getPackageManagerVersion(
+        'pnpm',
+        await lazyPgkJson.getValue()
+      );
+      const cleanedVersion = semver.coerce(pnpmVersionFromPackageJson);
+
+      if (cleanedVersion && semver.gte(cleanedVersion, '8.8.0')) {
```

Comment:
```suggestion
      if (semver.intersects(pnpmVersionFromPackageJson, '>=8.8.0')) {
```
as suggested in the discussion, probably hast a typo 🙃

Reply:
Thank you for the review. I have updated the code, and the current implementation no longer requires checking the `pnpm` version. Now it uses the `--config.ignore-scripts=true` flag with the `pnpm dedupe` command, which provides a more reliable solution across different `pnpm` versions. I've also updated the pull request title and description to reflect these changes. I appreciate your feedback and please let me know if there are any further adjustments needed.

<img width="741" alt="image" src="https://github.com/renovatebot/renovate/assets/38807139/6982b0db-5319-4d67-954e-c27656c56790">